console/workspaces/pages/eval/src/subComponents/EvaluatorForm.tsx

```
@@ -0,0 +1,1524 @@
+  // --- Code evaluator: auto-generated header tracking ---
+  const prevLevelRef = useRef(values.level);
+  const sourceRef = useRef(values.source);
+
+  // Keep sourceRef in sync (runs every render)
+  sourceRef.current = values.source;
+
+  // Clean up providers and validation listener on unmount
+  useEffect(() => {
+    return () => {
+      validationCleanupRef.current?.();
+      providerDisposablesRef.current.forEach((d) => d.dispose());
```

Biome (2.4.6)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)